### DISCUSSION TITLE
How can I write a simple cookie manager using JS interop using WASM?

### DISCUSSION THREAD

**Discussion by gpatrut-matica** [AvaloniaUI/Avalonia#17237] [Q&A]
Hello everyone,
I've been stuck for a while now trying to write a simple cookie manager from which I can get and set cookies. I've read the [documentation](https://learn.microsoft.com/en-us/aspnet/core/client-side/dotnet-interop/wasm-browser-app?view=aspnetcore-8.0) on how to use JSImport and JSExport but I can't seem to do more than just a simple alert or console.log and online there's not much information on this issue specifically...

I've tried a simple `document.cookie = cookie` but I get a very long error in the console without much information and I'm not even able to debug my code since breakpoints are never hit when building for browser...

Can anyone write me a very simple example, on how to define a simple function in JS that takes in a string and then call it from .NET code? 

Edit:

Sorry forgot some information, I am running Avalonia 11.1.3 in .NET 8 on a Windows 11 machine

**Comment by maxkatz6** [comment]
> I've tried a simple document.cookie = cookie but I get a very long error in the console without much information and I'm not even able to debug my code since breakpoints are never hit when building for browser...

Can you please share your code and an error?

**Comment by gpatrut-matica** [comment]
Nevermind sorry, after a day of sleep I have found that what I was missing was "globalThis" before the JS function I was trying to call...

Thanks and sorry for wasting your time!
